DDG4/python/DDSim/Helper/Geometry.py

```diff
@@ -36,45 +36,29 @@ def __init__(self):
     self._dumpDGDML_EXTRA = {"help": "If not empty, filename to dump the Geometry as GDML"}
     self.dumpGDML = ""
 
-    self._regexSDDetectorList = []
+    self._regexSDDict = {}
 
     self._closeProperties()
 
   @property
   def regexSensitiveDetector(self):
-    """Configure a sensitive detector for a given detector matched by regular expression (regex).
-
-    'Detector' and 'Match' are mandatory elements of the dictionary, other Keys are assigned as property to the object.
-
-    >>> SIM.geometry.regexSensitiveDetector = {'Detector': 'DRcalo',
-                                               'Match': ['(core|clad)'],
-                                               'OutputLevel': 3,
-                                              }
-
-    This can be assigned repeatedly to add multiple RegexSDs
+    """ The map key is the name of the Detector, and 'Match' is a mandatory elements of the dictionary, other Keys are
+        assigned as property to the object. OutputLevel _sets_ the outputlevel of the plugin, so lower numbers mean more
+        output from the plugin.
+
+        >>> SIM.geometry.regexSensitiveDetector['DRcalo'] = {
+                                                   'Match': ['(core|clad)'],
+                                                   'OutputLevel': 3,
+                                                  }
     """
-    return self._regexSDDetectorList
+    return self._regexSDDict
 
   @regexSensitiveDetector.setter
   def regexSensitiveDetector(self, val):
-    if not val:
-      return
     if isinstance(val, dict):
-      self.__checkRegexKeys(val)
-      self._regexSDDetectorList.append(val)
-    elif isinstance(val, list):
```

Comment:
When the ddsim steering file was parsed and values given to the main DD4hepSimulation instance, this caused the list we iterate over to be also appended to, which caused a a growth of the list until memory exhausted. Making sure that this doesn't happen was just causing code bloat, so I decided to change the interface of this. Should be OK since the feature was not yet in any tag.